common/appointment/create.ts

```diff
@@ -35,13 +36,41 @@ export abstract class AppointmentCreateGroupInput extends AppointmentCreateInput
     appointmentType: 'group';
 }
 
+const isAppointmentOneWeekLater = (datestring: Date) => {
+    const now = moment.now();
+    const start = moment(datestring);
+    const diffDays = start.diff(now, 'days');
+    if (diffDays >= 6) {
+        return true;
+    }
+    return false;
+};
+
+const isAppointmentFiveMinutesLater = (datestring: Date) => {
+    const now = moment.now();
+    const start = moment(datestring);
+    const same = start.isSame(now, 'date');
+    const after = start.isAfter(now);
+    if (same) {
+        const diff = start.diff(now, 'minutes');
+        if (diff >= 5) {
+            return true;
+        }
+    } else if (after) {
+        return true;
+    }
+
+    return false;
+};
+
 export const createMatchAppointments = async (matchId: number, appointmentsToBeCreated: AppointmentCreateMatchInput[]) => {
     const { pupil, student } = await prisma.match.findUniqueOrThrow({ where: { id: matchId }, include: { student: true, pupil: true } });
     const studentUserId = getUserIdTypeORM(student);
     const pupilUserId = getUserIdTypeORM(pupil);
+
     return await Promise.all(
-        appointmentsToBeCreated.map(
-            async (appointmentToBeCreated) =>
+        appointmentsToBeCreated.map(async (appointmentToBeCreated) => {
+            if (isAppointmentFiveMinutesLater(appointmentToBeCreated.start)) {
```

Comment:
This might result in creating some of the appointments, but not all appointments?
Another point is, that we only have to check the first one, if they are sorted correctly.